_transcripts/050-rust-1.56-1.57.md

```diff
@@ -0,0 +1,1302 @@
+adding things to the prelude, usually it doesn't take much ceremony. There are
+already namespacing rules in Rust that make it so that if, say, someone adds a
+thing to the prelude, and it clashes with a thing that you already defined in
+your own crate, then your own crate just wins. There's no real problem there.
+
+The problem with adding traits specifically though, is that it might introduce a
+method on a trait that clashes with one of your own methods on a similar named
+trait. And then that might make it so that calling a method might become
+ambiguous, which might stop code from compiling. You wouldn't get like a
+confusing method thing, it wouldn't, you know, at runtime, it wouldn't, you
+know, pick one at random, it would just say, hey, I can't compile any more, I'm
+not sure what you mean here, please use the extremely verbose explicit syntax to
+tell me which trait method you want to call in this case. And so that means that
+to add a trait to the prelude, you do need to do it over an edition. And so, the
+specific traits that have been added in the new edition are `TryInto`,
+`TryFrom`, and `FromIterator`. Jon, do you want to talk about those first two?
+
+__Jon__: Sure. So `TryInto` and `TryFrom` are, as their name implies, fallible
+versions of the `Into` and `From` trait. The idea here being that there are some
+types that can sometimes be converted into other types, but sometimes that
+conversion might fail. The obvious example here is parsing, right? So if you
+have a string, you can maybe turn it into a number, if it happens to contain a
+valid number, but it might also fail. You can think of `TryInto` and `TryFrom`
+as being the generic versions of something like the `FromStr` trait, which takes
+a string and— or takes any type that can be converted from a string fallibly.
+And `TryFrom` and `TryInto` were stabilized a long time ago, but it used to be
+that you explicitly bring them into scope in order to use them, which meant that
+people just didn't use them that much, or didn't even know that they existed.
+But now that they're in the prelude, it'll be a lot easier to just like type
+`.try_into()` or `.try_from()`, usually combined with a question mark operator,
+to do these conversions very straightforwardly.
+
+__Ben__: As for the last one there, `FromIterator`, it is one half of a pair. So
+`IntoIterator` is the other side of `FromIterator`. `IntoIterator` is the trait
+that controls how `for` loops work. And so if your type implements
+`IntoIterator`, then you can use it with for loops natively. `FromIterator` is
+kind of the opposite, where mostly you would only encounter `FromIterator` as
+part of the `collect` method. So if you have an iterator chain of things, and
+then you call `collect()` on the end, under the hood that is using
+`FromIterator` to do all the magic. And `collect` (_unintelligible— 12:21_) very
+often and so, but you don't need `FromIterator` in scope just to collect things.
```

Comment:
Oops, the actual timestamp here is 18:51.

Reply:
I can't piece it together either. I think that's okay though — the point about not needing `FromIterator` in scope is the critical one.